combinations.h

```diff
@@ -539,7 +539,8 @@ std::uintmax_t
 count_each_permutation(BidirIter first, BidirIter mid, BidirIter last)
 {
     return count_each_permutation<std::uintmax_t>
-                          (std::distance(first, mid), std::distance(mid, last));
+                          (static_cast<std::uintmax_t>(std::distance(first, mid)),
+			   static_cast<std::uintmax_t>(std::distance(mid, last)));
```

Comment:
I believe that with your change we could remove the explicit instantiation on line 541:
```c++
return count_each_permutation
```

but I haven't tested that suggestion.

Reply:
That does compile for me as well.   I've done a force push with that fix.